File: app/src/main/java/workshop/akbolatss/tagsnews/screen/news/NewsSource.java
package workshop.akbolatss.tagsnews.screen.news;

import android.support.v4.app.Fragment;

import workshop.akbolatss.tagsnews.screen.recommendations.RecommendationsFragment;

public class NewsSource {

    private String mName;
    private String mUrl;
    private Fragment mFragment;
    private boolean isRecommendations;

    public NewsSource(String mName, String mUrl, boolean isRecommendations) {
        this.mName = mName;
        this.mUrl = mUrl;
        this.isRecommendations = isRecommendations;
        if (isRecommendations) {
            mFragment = new RecommendationsFragment();
        } else {
            mFragment = new NewsFragment();
        }
    }

    public String getName() {
        return mName;
    }

    public String getUrl() {
        return mUrl;
    }

    public Fragment getFragment() {
        return mFragment;
    }

    public boolean isRecommendations() {
        return isRecommendations;
    }
}

File: app/src/main/java/workshop/akbolatss/tagsnews/screen/news/NewsSource.kt
package workshop.akbolatss.tagsnews.screen.news

import android.support.v4.app.Fragment

import workshop.akbolatss.tagsnews.screen.recommendations.RecommendationsFragment

class NewsSource(val name: String, val url: String?, private val isRecommendations: Boolean) {
    var fragment: Fragment
        private set

    init {
        fragment = if (isRecommendations) {
            RecommendationsFragment()
        } else {
            NewsFragment()
        }
    }
}
